invesalius/gui/preferences.py

```diff
@@ -577,101 +557,378 @@ def __init__(self, parent, navigation, tracker, pedal_connector, neuronavigation
+            elif (obj_tracker_id := coil_registration["tracker_id"]) != self.tracker.tracker_id:
+                wx.MessageBox(
+                    _(
+                        f"Cannot select this coil, its tracker [{const.TRACKERS[obj_tracker_id - 1]}] does not match the selected tracker [{const.TRACKERS[self.tracker.tracker_id - 1]}]"
+                    ),
+                    _("InVesalius 3"),
+                )
+                self.coil_btns[name][0].SetValue(False)  # Unpress the button
+                return
+
+            # Press the coil button here in case selection was done via code without pressing button
+            self.coil_btns[name][0].SetValue(True)
+
+        # Select/Unselect coil
+        Publisher.sendMessage("Select coil", coil_name=name, coil_registration=coil_registration)
+
+        n_coils_selected = len(navigation.coil_registrations)
+        n_coils = navigation.n_coils
+
+        # Update labels telling which coil is selected (for single coil mode) and how many coils to select (for multicoil mode)
+        self.config_txt.SetLabel(
+            f"{os.path.basename(self.coil_registrations.get('default_coil', {}).get('path', 'None'))}"
+        )
+        self.sel_sizer.GetStaticBox().SetLabel(
+            f"TMS coil selection ({n_coils_selected} out of {n_coils})"
+        )
+
+        # Update robot coil combobox
+        if self.choice_robot_coil is not None:
+            self.choice_robot_coil.Set(list(navigation.coil_registrations))
+            self.choice_robot_coil.SetStringSelection(self.robot.GetCoilName() or "")
+
+        if n_coils_selected == n_coils:
+            self.CoilSelectionDone()
+        else:  # Enable all buttons
+            Publisher.sendMessage("Coil selection done", done=False)
+            for btn, *junk in self.coil_btns.values():
+                btn.Enable(True)
+
+    def OnRightClickCoil(self, event, name):
+        def DeleteCoil(event, name):
+            # Unselect the coil first
+            self.OnSelectCoil(name, select=False)
+            del self.coil_registrations[name]
+
+            # Remove the coil-button
+            self.coil_btns[name][0].Destroy()
+            del self.coil_btns[name]
+
+            # Remove the coil from the config file
+            self.session.SetConfig("coil_registrations", self.coil_registrations)
+
+            # Remove coil from navigation and CoilVisualizer
+            Publisher.sendMessage("Select coil", coil_name=name, coil_registration=None)
+
+        menu = wx.Menu()
+        delete_coil = menu.Append(wx.ID_ANY, "Delete coil")
+        save_coil = menu.Append(wx.ID_ANY, "Save coil to OBR file")
+
+        self.Bind(wx.EVT_MENU, (lambda event, name=name: DeleteCoil(event, name)), delete_coil)
+        self.Bind(
+            wx.EVT_MENU,
+            (lambda event, name=name: self.OnSaveCoilToOBR(event, coil_name=name)),
+            save_coil,
+        )
+        self.PopupMenu(menu, event.GetPosition())
```

Comment:
You can remove event.GetPosition(). That will make the menu appears where is clicked